pkg/process/monitor/process_monitor.go

```diff
@@ -188,10 +190,14 @@ func (pm *ProcessMonitor) mainEventLoop() {
 			switch ev := event.Msg.(type) {
 			case *netlink.ExecProcEvent:
 				pm.execCount.Inc()
-				pm.handleProcessExec(int(ev.ProcessPid))
+				if pm.hasExecCallbacks.Load() {
```

Comment:
Please add a comment about why this boolean is needed (an optimization issue)

Reply:
done